src/adt/test_utils.rs

```diff
@@ -113,43 +117,59 @@ where
     );
 }
 
+fn u128_to_array<const WORD_LENGTH: usize>(u: u128) -> [u8; WORD_LENGTH] {
+    let mut bytes = [0u8; WORD_LENGTH];
+    bytes[..16].copy_from_slice(&u.to_be_bytes());
+    bytes
+}
+
+fn word_to_array<const WORD_LENGTH: usize>(word: [u8; WORD_LENGTH]) -> u128 {
+    let mut bytes = [0; 16];
+    bytes.copy_from_slice(&word[..16]);
+    u128::from_be_bytes(bytes)
+}
+
 /// Tests concurrent guarded write operations on a Memory ADT implementation.
 ///
 /// Spawns multiple threads to perform concurrent counter increments.
 /// Uses retries to handle write contention between threads.
 /// Verifies the final counter matches the total number of threads.
-pub async fn test_guarded_write_concurrent<Memory>(memory: &Memory, seed: [u8; 32])
-where
+pub async fn test_guarded_write_concurrent<const WORD_LENGTH: usize, Memory>(
+    memory: &Memory,
+    seed: [u8; KEY_LENGTH],
+) where
     Memory: 'static + Send + Sync + MemoryADT + Clone,
-    Memory::Address: Send + From<[u8; 16]>,
-    Memory::Word: Send + Debug + PartialEq + From<[u8; 16]> + Into<[u8; 16]> + Clone + Default,
+    Memory::Address: Send + From<[u8; ADDRESS_LENGTH]>,
+    Memory::Word:
+        Send + Debug + PartialEq + From<[u8; WORD_LENGTH]> + Into<[u8; WORD_LENGTH]> + Clone,
     Memory::Error: Send + std::error::Error,
+    [(); WORD_LENGTH]: Sized,
```

Comment:
Why is this bound necessary ? Isn't it always the case ?

Reply:
indeed, removed.